### ISSUE TITLE
Training the model

### ISSUE THREAD

**Issue by veroveroxie** [LifuWang-66/DistillT5#6]
Hi, thanks for your great paper and code.

I am trying to run your training code. I trained the flan-t5-base with 5000 iterarations, but it generates same images for every prompt. I know that you have trained the model for more iterations, I am wondering if there is any iteration so that model suddenly understands the prompt or did you apply any pretraining or loss regularization to the new encoder?

<img width="1213" height="205" alt="Image" src="https://github.com/user-attachments/assets/34d6c4e8-d1ad-4715-b81d-9136f3d3a69c" />

Following your suggestion, I only use T2IComp, but I have to use batch size = 1x8 = 8 instead of 32.

`accelerate launch train_flux.py  \
    --train_text_encoder \
    --mixed_precision bf16 \
    --train_batch_size 1 \
    --resolution 512 \
    --text_encoder_lr 1e-4 \
    --compbench_path data/T2I-CompBench \
    --num_train_epochs 1`

Could you please share any thoughts?

Thanks

**Comment by LifuWang-66** [comment]
Thanks for reaching out! If you are training from scratch, then maybe you should train for at least several epochs on the T2IComp dataset (as it is very small). We trained for 50000 iterations on the dataset, so I'd expect a higher number of iterations in your case. If the images looked better than the ones without training, then you should be on the right track.

**Comment by johnr14** [comment]
Hmm, I think that a bigger and more representative dataset could also help. **1M** samples should help !

Look at [fancyfeast/joy-captioning-20250328b](https://huggingface.co/datasets/fancyfeast/joy-captioning-20250328b) and train on the `answer` column. Could filter for `caption` in `question_type` to remove tags and slim down the dataset a bit.

Be advised that it is uncensored and could have quite explicit content:
> Uncensored: Equal coverage of SFW and NSFW concepts. No "cylindrical shaped object with a white substance coming out on it" here.

> Diversity: All are welcome here. Do you like digital art? Photoreal? Anime? Furry? JoyCaption is for everyone. Pains are being taken to ensure broad coverage of image styles, content, ethnicity, gender, orientation, etc.

> Minimal Filtering: JoyCaption is trained on large swathes of images so that it can understand almost all aspects of our world. almost. Illegal content will never be tolerated in JoyCaption's training.
